cdc-accelerator/core/sitesCache.js

```diff
@@ -40,4 +38,27 @@ export default class SitesCache {
     static getSiteInfo(_apiKey) {
         return SitesCache.cache.find(({ apiKey }) => apiKey === _apiKey)
     }
+
+    static isCacheUpdated(configuration) {
+        if (!configuration?.cache) {
+            return false
+        }
+        let updated = true
+        if (configuration.source) {
+            configuration.source.forEach((entry) => {
+                updated &= configuration.cache.some(({ apiKey }) => {
+                    return entry.apiKey === apiKey
+                })
+            })
+        }
+
+        if (updated && configuration.deploy) {
+            configuration.deploy.forEach((entry) => {
+                updated &= configuration.cache.some(({ apiKey }) => {
+                    return entry.apiKey === apiKey
+                })
+            })
+        }
```

Comment:
The code inside `if (configuration.source) {` seems to be duplicated in `if (updated && configuration.deploy) {`. We can create a separate function for that specific part.

Reply:
done